### PR TITLE
feat(data-transfer): support transfer mysql data by DataX

### LINKED ISSUE #247: [Feature]: Support import and export MySQL datasource
### Is your feature request related to a problem?

ODC currently supports native MySQL datasource, but does not support importing and exporting MySQL. 
This is an important feature that we hope ODC can support.

### Describe the solution you'd like

Similar to the import and export of OceanBase datasource, we hope to support the following functions:
- Import structure; 
- Export structure; 
- Import data; 
- Export data; 
- Import and export of Single table ; 
- Import and export of the entire database.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by LuckyPickleZZ**
#### What type of PR is this?
type-feature

#### What this PR does / why we need it:

In this PR, we can finally support transferring data for MySQL database.  Here are some review tips.
1. It's implemented by datax, which we import by maven system dependency. The jar's relative path is `import/datax-core-0.0.1-SNAPSHOT.jar`. When we execute 'mvn package' , this jar would be included in 'task-plugin-mysql-xxx.jar'.
2. The most transferring works are handled in `DataXTransferJob`. Each of `DataXTransferJob` can only transfer ONE object once.
3. DataX requires jsonfile type or map as job `Configuration`. Those classes under `com/oceanbase/odc/plugin/task/mysql/datatransfer/job/datax/model` are to generate `Configuration` in `ConfigurationResolver`. The files in the `resources` are also used for this .

#### Which issue(s) this PR fixes:

Fixes #247

### CODE REVIEW

File: server/odc-server/src/main/resources/log4j2.xml [hunk 1]
```diff
@@ -106,6 +106,28 @@
             </Routes>
         </Routing>
 
+        <Routing name="DataXInfoRoutingAppender">
```
Comment (LuckyPickleZZ) [inline server/odc-server/src/main/resources/log4j2.xml:109]: This config is to export DataX's log into datax.all. This log will not be shown in front-end. It's only for troubleshooting issues.

File: server/plugins/task-plugin-mysql/src/main/assembly/assembly.xml [hunk 2]
```diff
@@ -12,4 +12,12 @@
...
         </fileSet>
     </fileSets>
+    <dependencySets>
+        <dependencySet>
+            <outputDirectory>/</outputDirectory>
+            <useProjectArtifact>true</useProjectArtifact>
+            <unpack>true</unpack>
+            <scope>system</scope>
+        </dependencySet>
+    </dependencySets>
```
Comment (LuckyPickleZZ) [inline server/plugins/task-plugin-mysql/src/main/assembly/assembly.xml:22]: This is to package datax-core-0.0.1-SNAPSHOT.jar into task-plugin-mysql.jar

File: server/plugins/task-plugin-mysql/src/main/java/com/oceanbase/odc/plugin/task/mysql/datatransfer/job/datax/DataXTransferJob.java [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+    public void run() throws Exception {
+
+        DataXJobMonitor monitor = new DataXJobMonitor();
+
+        try {
+            Configuration configuration = ConfigurationResolver.resolve(jobConfig);
+            configuration.set(CoreConstant.DATAX_CORE_CONTAINER_JOB_ID, jobId);
+            this.containerCommunicator = new StandAloneJobContainerCommunicator(configuration);
+
+            new Thread(monitor, "DataX-Monitor-" + Thread.currentThread().getName()).start();
```
Comment (yhilmare) [inline server/plugins/task-plugin-mysql/src/main/java/com/oceanbase/odc/plugin/task/mysql/datatransfer/job/datax/DataXTransferJob.java:81]: how to stop it? prefer not to new a thread directly~

Comment (LuckyPickleZZ) [reply server/plugins/task-plugin-mysql/src/main/java/com/oceanbase/odc/plugin/task/mysql/datatransfer/job/datax/DataXTransferJob.java:81]: This monitor will be stopped in finally block by calling it's stop method. In the `stop`, a stop flag will be set into true to break loop.

```JAVA
        DataXJobMonitor monitor = new DataXJobMonitor();
        try {
           ...
            new Thread(monitor, "DataX-Monitor-" + Thread.currentThread().getName()).start();
           ...
        } finally {
            monitor.stop();
        }

```

**Review by smallsheeeep — COMMENTED** [review]
Marvelous 👍🏻

**Review by smallsheeeep — COMMENTED** [review]
Are we giving up on data masking in MySQL mode? Will this solution  support data masking in the future? @yhilmare @yizhouxw

File: server/plugins/task-plugin-mysql/src/main/java/com/oceanbase/odc/plugin/task/mysql/datatransfer/job/datax/DataXTransferJob.java [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+        }
+        FileUtils.write(file, JsonUtils.toJson(ImmutableMap.of("job", jobConfig)), StandardCharsets.UTF_8);
+        return file;
+    }
+
+    private String[] buildDataXExecutorCmd(String dataxHomePath, String tmpFilePath) {
+        List<String> command = new ArrayList<>();
+        command.add("java");
+        command.add("-server");
+        command.add("-Xms1g");
```
Comment (yhilmare) [inline server/plugins/task-plugin-mysql/src/main/java/com/oceanbase/odc/plugin/task/mysql/datatransfer/job/datax/DataXTransferJob.java:130]: make this configurable

**Review by yhilmare — APPROVED** [review]
lgtm

File: server/plugins/task-plugin-mysql/src/main/java/com/oceanbase/odc/plugin/task/mysql/datatransfer/job/datax/DataXTransferJob.java [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+        command.add("-1");
+        command.add("-job");
+        command.add(tmpFilePath);
+        return command.toArray(new String[0]);
+    }
+
+    private void analysisStatisticsLog(InputStream inputStream) throws IOException {
+        try (BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream, StandardCharsets.UTF_8))) {
+            String line;
+            while ((line = reader.readLine()) != null && !Thread.currentThread().isInterrupted()) {
```
[the remark this answers is not in the document]
Comment (LuckyPickleZZ) [reply server/plugins/task-plugin-mysql/src/main/java/com/oceanbase/odc/plugin/task/mysql/datatransfer/job/datax/DataXTransferJob.java]: sure, done

File: server/plugins/task-plugin-mysql/src/main/java/com/oceanbase/odc/plugin/task/mysql/datatransfer/job/datax/DataXTransferJob.java (hunk 4 above)
Comment (LuckyPickleZZ) [reply server/plugins/task-plugin-mysql/src/main/java/com/oceanbase/odc/plugin/task/mysql/datatransfer/job/datax/DataXTransferJob.java]: done

**Pull request merged**